SIPS/sip-29.md

```diff
@@ -243,9 +267,126 @@ type FungibleAssetMetadata = {
   // List of asset units.
   units: FungibleAssetUnit[];
 };
+```
+### Asset Market Data
 
-// Represents the metadata of an asset.
-type AssetMetadata = FungibleAssetMetadata
+```typescript
+type FungibleAssetMarketData = {
+  // Represents a fungible asset market data.
+  fungible: true;
+
+  // The market cap of the asset represented as a decimal number in a string.
+  marketCap?: string;
+
+  // The total volume of the asset represented as a decimal number in a string.
+  totalVolume?: string;
+
+  // The circulating supply of the asset represented as a decimal number in a string.
+  circulatingSupply?: string;
+
+  // The all time high of the asset represented as a decimal number in a string.
+  allTimeHigh?: string;
+
+  // The all time low of the asset represented as a decimal number in a string.
+  allTimeLow?: string;
+
+  pricePercentChange?: {
+    // The `all` value is a special interval that represents all available data.  
+    all?: number;  
+
+    // The interval key MUST follow the ISO 8601 duration format.  
+    [interval: string]: number;
+  };
+};
+```
+
+## Appendix II: Non-fungible Asset Metadata & Market Data
+
+The following asset metadata fields for a non-fungible asset are defined.
+
+### Asset Metadata
+```typescript
+type NonFungibleAssetCollection = {
+  // Human-friendly name of the asset collection.
+  name: string;
+
+  // Ticker symbol of the asset collection.
+  symbol: string;
+
+  // The number of tokens in the collection.
+  tokenCount: string;
+
+  // The creator address of the asset.
+  creator?: Caip19AssetType;
+
+  // Base64 data URI or URL representation of the asset icon.
+  imageUrl: string;
+};
+
+type NonFungibleAssetMetadata = {
+  // Human-friendly name of the asset.
+  name?: string;
+
+  // Ticker symbol of the asset.
+  symbol?: string;
+
+  // Base64 data URI or URL representation of the asset image.
+  imageUrl: string;
+
+  // The description of the asset.
+  description?: string;
+
+  // Represents a non-fungible asset
+  fungible: false;
+
+  // Attributes of the non-fungible asset.
+  attributes?: Record<string, string | number>;
+
+  // The collection of the asset.
+  collection: NonFungibleAssetCollection;
+};
+```
+
+### Asset Market Data
+```typescript
+type NonFungibleAssetMarketData = {
+  // Represents a non-fungible asset market data.
+  fungible: false;
+
+  // The last sale of one asset in the collection.
+  lastSale?: {
+    // The asset that was sold.
+    asset: Caip19AssetTypeOrId;
+    // The price at which is was sold represented as a decimal number in a string.
+    amount: string;
+  };
+
+  // The top bid on the asset.
```

Comment:
I like that all attributes are optional in `NonFungibleAssetMarketData` 🙏 
In case we have new marketData coming from NFT-API, will we have to update the snap first?

Reply:
From the types you sent me it seems like all of these can be optional :)

If we add new data from the NFT-API we will have to update the platform if it's not yet specified here and then update the snap yes !